lib/improver/blending/weighted_blend.py

```diff
@@ -539,60 +861,21 @@ def process(self, cube, weights=None):
             raise ValueError('Blending coordinate {} has no associated '
                              'dimension'.format(self.coord))
 
-        # Ensure input cube is ascending along blending coordinate, so that
-        # weights match cube slices
+        # Ensure input cube and weights cube are ordered equivalently along
+        # blending coordinate.
         cube = sort_coord_in_cube(cube, self.coord, order="ascending")
+        if weights is not None:
+            weights = sort_coord_in_cube(weights, self.coord,
+                                         order="ascending")
```

Comment:
As discussed: somewhere around here, could we put in an explicit check that the weights cube and data cube have the same (named) coordinate dimensions in the same order?

Reply:
Not here. The weights cube might well be expected to have only one coordinate (e.g. a 1D array of weights). We can check that these are over the ```self.coord```, where this is the blending coordinate.

We can check that in the case that a multi-dimensional cube is provided that this has the same coordinates as the data cube, but we will have to allow for the percentile coordinate to be missing.

I'll take a look at this.

Reply:
I've added a test here to ensure that at least the blending coordinate is in the weights cube.
To check the multidimensional weights case I have added a check in the shape_weights function. The reason to add it here is that the percentile coordinate has already been removed from the diagnostic cube, which makes the test straightforward.

We need to be aware that I have assumed we will never want to make weights cubes that include a explicit percentile coordinate. This assumption is founded on the belief that we will not want to vary weights between percentiles, e.g. won't want the 10th percentile weight at time T-0 to be different to the 90th percentile weight at time T-0.

Producing cubes that vary with percentile would require much larger cubes, and these would have to be rebuilt if the number of percentiles being generated is changed. The code as it stands would not handle these.

Reply:
Agreed.